integration/nwo/common/docker/utils.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package docker
+
+import (
+	"net"
+	"strings"
+	"sync"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	"github.com/pkg/errors"
+)
+
+var logger = flogging.MustGetLogger("fsc.integration.fabric")
+
+type Docker struct {
+	Client *docker.Client
+}
+
+var singleInstance *Docker
+var lock = &sync.Mutex{}
+
+func GetInstance() (*Docker, error) {
+
+	if singleInstance == nil {
```

Comment:
this check should happen when the lock has been acquired already, otherwise you can have more than one thread after the if.

Reply:
good catch

Reply:
but did you fix it?